db/sqlinterfaces.c

```diff
@@ -292,6 +292,31 @@ struct sqlclntstate *get_sql_clnt(void){
   return thd->clnt;
 }
 
+static pthread_key_t dispatch_clnt_key;
```

Comment:
How about we turn this into a __thread struct sqlclntstate *clnt_ptr = NULL;
and dereference accordingly since it can be faster to work with than pthread keys.

Reply:
I like this - `bdb_temp_table_create_type` should take a flags param instead of checking for various unrelated things in the temp-table code.